source/ast/expressions/AssertionExpr.cpp

```diff
@@ -523,22 +625,44 @@ void SimpleAssertionExpr::requireSequence(const ASTContext& context, DiagCode co
     }
 }
 
-bool SimpleAssertionExpr::admitsEmptyImpl() const {
+bitmask<NondegeneracyStatus> SimpleAssertionExpr::checkNondegeneracyImpl() const {
+    if (expr.kind == ExpressionKind::AssertionInstance) {
+        auto& aie = expr.as<AssertionInstanceExpression>();
+        if (aie.type->isSequenceType())
+            return aie.body.checkNondegeneracy();
+    }
+
+    bitmask<NondegeneracyStatus> res = NondegeneracyStatus::None;
+
+    // If simple sequence expression can be evaluated to `false` (`0`)
+    // manually then it should be reported as sequence
+    // with no possible match
+    res = (isNullExpr) ? res | NondegeneracyStatus::AdmitsNoMatch : res;
+
     if (repetition) {
         auto result = repetition->admitsEmpty();
-        if (result == SequenceRepetition::AdmitsEmpty::Yes)
-            return true;
-        if (result == SequenceRepetition::AdmitsEmpty::No)
-            return false;
+        res = (result == SequenceRepetition::AdmitsEmpty::Yes)
```

Comment:
This seems unnecessarily clunky; you can just check `if (result == SequenceRepetition::AdmitsEmpty::Yes)` and add the flag if so.

Reply:
fixed